core/field_dropdown.js

```diff
@@ -141,6 +141,13 @@ Blockly.FieldDropdown.prototype.imageElement_ = null;
  */
 Blockly.FieldDropdown.prototype.imageJson_ = null;
 
+/** 
+ * Height of the selected item to adjust the scroll position.
+ * @type {number}
+ * @private
+ */
+Blockly.Field.SELECTED_ITEM_Y_POS = 0;
```

Comment:
This should go on the instance of the field. 
Instead, try storing the selected menu item in ``widgetCreate_`` into an instance property. Perhaps ``this.selectedMenuItem_``, and then use the built in utility method: ``Blockly.utils.style.scrollIntoContainerView`` to scroll to that item, after render.